src/cli.ts

```diff
@@ -128,17 +134,34 @@ const run = async () => {
   }
 
   const fileNames = await resolvePrismaFiles(args);
-  const fileViolationList = await lintPrismaFiles({
+  const fileResults = await lintPrismaFiles({
     rules,
     fileNames,
   });
 
-  outputToConsole(fileViolationList, outputFormat, quiet);
+  outputToConsole(fileResults, outputFormat, quiet);
 
-  const hasViolations = fileViolationList.some(
+  const hasViolations = fileResults.some(
     ({ violations }) => violations.length > 0,
   );
   if (hasViolations) {
+    if (options.fix) {
+      // eslint-disable-next-line no-console
+      console.log('\nAttempting to automatically fix violations...');
+      for (const fileResult of fileResults) {
+        const fixedCount = applyFixesToPrismaSchema(fileResult);
+        if (fixedCount > 0) {
+          await overwriteSourceCode(fileResult);
+        }
+        // eslint-disable-next-line no-console
+        console.log(
+          `Fixed ${fixedCount} ${pluralize('violations', fixedCount)} in ${
+            fileResult.fileName
+          }`,
+        );
+      }
+      // TODO - run in a loop until there are no violations?
```

Comment:
im not sure i understand why this would be necessary.

i'd imagine the most efficient way to do this is to iterate over the `fileResults` array and keep track of how many errors there area and how many are unfixable for logging

Reply:
gotcha, probably makes sense then. otherwise we could have a false idea of no issues